app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt

```diff
@@ -18,18 +17,20 @@ class DataSyncTest {
 
   @Test
   fun `when syncing everything, all the model syncs should be invoked`() {
-    val modelSync1 = mock<ModelSync>()
-    val modelSync2 = mock<ModelSync>()
-    val modelSync3 = mock<ModelSync>()
-
-    whenever(modelSync1.syncConfig()).thenReturn(createSyncConfig(SyncGroup.DAILY))
-    whenever(modelSync1.sync()).thenReturn(Completable.complete())
+    val modelSync1 = FakeModelSync(
```

Comment:
Mocking or creating a fake class would do the same thing, right?

Reply:
The problem with mocks is that I need to keep my mocks updated when changing implementation. Creating a fake object allows me to change implementation without needing to update the fake as much.

Reply:
Should'nt the mock be updating itself when we change the implementation of the class?

Reply:
We might get compilation errors in some cases, and other times we may not. Plus, since mocks need you to provide the stub each time, there might be scenarios where a test is passing because the mock stub is not behaving the same as a change we introduced, which gives a false sense of security.